### DISCUSSION TITLE
importing .msh files and setting boundary conditions

### DISCUSSION THREAD

**Discussion by jakpatel** [idaholab/moose#23129] [Q&A Meshing] [answered]
Hi!

I'm trying to setup a simple square mesh for the Laplace Young problem in the tutorials using gmsh. When I run the code without any boundary conditions, it runs fine but as soon as I try to set BCs, it keeps telling me:

 (BCs/left/boundary):  the following node sets (ids) do not exist on the mesh: left (1)

I have a very simple 4-element mesh currently but I've been stuck on this for a while and I don't know how to proceed. Any help is very highly appreciated. 

Thanks,
J

.msh:
```
$MeshFormat
2.2 0 8
$EndMeshFormat
$PhysicalNames
4
1 1 "left"
1 2 "right"
1 3 "top"
1 4 "bottom"
$EndPhysicalNames
$Nodes
4
1 0 0 0
2 0 1 0
3 1 1 0
4 1 0 0
$EndNodes
$Elements
4
1 1 2 4 1 1 4
2 1 2 2 2 4 3
3 1 2 3 3 3 2
4 1 2 1 4 2 1
$EndElements


.geo:

//+
Point(1) = {0, 0, 0, 1.0};
//+
Point(2) = {0, 1, 0, 1.0};
//+
Point(3) = {1, 1, 0, 1.0};
//+
Point(4) = {1, 0, 0, 1.0};
//+
Point(5) = {1, 0, 0, 1.0};
//+
Line(1) = {1, 4};
//+
Line(2) = {4, 3};
//+
Line(3) = {3, 2};
//+
Line(4) = {2, 1};
//+
Curve Loop(1) = {4, 1, 2, 3};
//+
Plane Surface(1) = {1};
//+
Physical Curve("left") = {4};
//+
Physical Curve("right") = {2};
//+
Physical Curve("top") = {3};
//+
Physical Curve("bottom") = {1};
```
MOOSE Input:
```
[Mesh]
  file = 'squarev2.msh'
[]
construct_side_list_from_node_list=true
[Variables]
  [u]
  []
[]

[Kernels]
  [diffusion]
    type = LaplaceYoungDiffusion
    variable = u
  []
  [source]
  	type = LaplaceYoungSource
  	variable = u
  	kappa = 0.9
  []
[]

[BCs]
  [left]
    type = DirichletBC
    variable = u
    value = 0.2
    boundary = "left"
  []
[]

[Executioner]
  type = Steady
  solve_type = Newton
  petsc_options_iname = '-pc_type'
  petsc_options_value = 'hypre'
[]

[Outputs]
  vtk = true
[]
```

**Comment by GiudGiud** [comment, marked as answer]
Hello

Should this be a 2D mesh? It looks 1D (multiple 1D segments). If it's 1D then a boundary is not a segment, it's a point

Guillaume

**Reply by jakpatel** [reply]
Hi GuidGuid, thanks for your reply. I'm basically just trying to make a square. I did a lot of different permutations on the input file. This was the simplest one. But the idea is to draw a 2D square, discretize it, and then have moose read it and run a problem. I eventually need to be able to read in large unstructured grids with multiple materials scattered randomly in the system but I'm currently not even able to do a square. Help?

**Reply by GiudGiud** [reply]
This is a problem with your gmsh mesh though. You are not making a  filled square here, just the outer boundary.

If you need a square mesh
https://mooseframework.inl.gov/source/meshgenerators/CartesianMeshGenerator.html

**Reply by jakpatel** [reply]
I refined the mesh slightly:

$MeshFormat
2.2 0 8
$EndMeshFormat
$PhysicalNames
4
1 1 "left"
1 2 "right"
1 3 "top"
1 4 "bottom"
$EndPhysicalNames
$Nodes
8
1 0 0 0
2 0 1 0
3 1 1 0
4 1 0 0
5 0.4999999999986718 0 0
6 1 0.4999999999986718 0
7 0.5000000000013305 1 0
8 0 0.5000000000013305 0
$EndNodes
$Elements
8
1 1 2 4 1 1 5
2 1 2 4 1 5 4
3 1 2 2 2 4 6
4 1 2 2 2 6 3
5 1 2 3 3 3 7
6 1 2 3 3 7 2
7 1 2 1 4 2 8
8 1 2 1 4 8 1
$EndElements
[sqr_mesh_ex.pdf](https://github.com/idaholab/moose/files/10386957/sqr_mesh_ex.pdf)

**Reply by GiudGiud** [reply]
It still reads as a 1D multi-segment mesh in MOOSE
I m not sure what's wrong, I dont use gmsh.

**Reply by jakpatel** [reply]
No worries! I'll keep trying and see if I'm doing something wrong with gmsh. Plmk if you think of a workaround. Thanks! :)

**Reply by GiudGiud** [reply]
@grmnptr do you see why his mesh isnt 2D ?
